File: content/snippet/code-signal/arcade/intro/eruption-of-light/is-beautiful-string/is-beautiful-string.js
function isBeautifulString(inputString) {
  const charMap = {};
  const charCode = 'a'.charCodeAt(0);
  for (let i = 0; i < 26; i++) {
    charMap[String.fromCharCode(charCode + i)] = 0;
  }
  for (let i = 0; i < inputString.length; i++) {
    charMap[inputString[i]]++;
  }
  const charMapKeys = Object.keys(charMap);
  for (let i = 0; i < charMapKeys.length - 1; i++) {
    if (charMap[charMapKeys[i]] < charMap[charMapKeys[i + 1]]) {
      return false;
    }
  }
  return true;
}

console.log(isBeautifulString('bcdeff'));
console.log(isBeautifulString('aabb'));
console.log(isBeautifulString('bbc'));

File: content/snippet/code-signal/arcade/intro/eruption-of-light/is-beautiful-string/is-beautiful-string.py
import string


def is_beautiful_string(input_string):
    count_dict = sorted([
        (k, input_string.count(k))
        for k in set((input_string))
    ])

    if(count_dict[0][0] != 'a'):
        return False

    for i in range(len(count_dict) - 1):
        if(
            ord(count_dict[i+1][0]) - ord(count_dict[i][0]) != 1
            or
            count_dict[i+1][1] > count_dict[i][1]
        ):
            return False

    return True


print(is_beautiful_string('bcdeff'))
print(is_beautiful_string('aabb'))
